### PR TITLE
docs: Update Readme

### PR DESCRIPTION
**Pull request by sithlord48**
Update the readme ONLY [Preview](https://github.com/deskflow/deskflow/blob/readme_update/README.md)
Split other changes into #7791

### CODE REVIEW

**Comment by nbolton** [comment]
> * mv BUILD.MD -> doc/build.md
> * mv CONTRIBUTING.md -> doc/contributing.md

Not a fan of all caps rage, eh? :smile:

File: doc/build.md [hunk 1]
```diff
@@ -1,4 +1,32 @@
-# Build Deskflow
+<!--
+    SPDX-FileCopyrightText: 2024 Symless Ltd.
```
[the remark this answers is not in the document]
Comment (nbolton) [reply doc/build.md:2]: This is the way

[the remark this answers is not in the document]
Comment (nbolton) [reply doc/build.md:2]: This is the way

File: doc/server-configs.md
Comment (nbolton) [inline doc/server-configs.md:1]: @sithlord48 I renamed the file to `doc/server-configs.md` and added more context around how server config files work. I also made the example computer names consistent (moe, larry, and curly) and fixed the ASCII art (which was completely wrong in the originals).

**Comment by nbolton** [comment]
@sithlord48 Ready to land this? If not, what's left?

File: doc/server-configs.md
Comment (nbolton) [inline doc/server-configs.md:1]: @sithlord48 I renamed the file to `doc/server-configs.md` and added more context around how server config files work. I also made the example computer names consistent (moe, larry, and curly) and fixed the ASCII art (which was completely wrong in the originals).

File: doc/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  set(DOXYGEN_STRIP_FROM_PATH ${CMAKE_SOURCE_DIR})
+  set(DOXYGEN_USE_MDFILE_AS_MAINPAGE mainpage.md)
+  set(DOXYGEN_QUIET YES)
+
+  doxygen_add_docs(docs ${CMAKE_SOURCE_DIR}/doc ${CMAKE_SOURCE_DIR}/src
+                   COMMENT "Generating Documentation" ALL)
+
+  #This is only to show files used by the documentation
+  #within the project tree
+  target_sources(docs PRIVATE build.md contributing.md configuration.md
```
Comment (sithlord48) [inline doc/CMakeLists.txt:18]: can we get these one per line please

File: doc/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                              mainpage.md)
+
+  install(
+    DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/html"
+    DESTINATION ${CMAKE_INSTALL_DOCDIR}
+    COMPONENT deskflow_docs)
```
Comment (sithlord48) [inline doc/CMakeLists.txt:24]: closing ) on its own line in line with install

File: doc/CMakeUserPresets.json [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    },
+    {
+      "name": "windows",
+      "inherits": "base",
+      "hidden": true,
+      "generator": "Ninja",
+      "cacheVariables": {
+        "CMAKE_C_COMPILER": "cl.exe",
+        "CMAKE_CXX_COMPILER": "cl.exe",
+        "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
```
Comment (sithlord48) [inline doc/CMakeUserPresets.json:21]: fwiw this breaks my local build on windows where i have the cache outside the build dir.. 
windows uses on cli may need to set this path for their machine specificity

File: doc/config.toml.example [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# Example file for `--config-toml` arg.
```
Comment (sithlord48) [inline doc/config.toml.example:1]: this should be in the `configuration.md` page as a code block  like the other config examples.

File: doc/server-configs.md
Comment (sithlord48) [reply doc/server-configs.md:1]: did you not update this link ?

File: doc/mainpage.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+OUR Documents main page
```
Comment (sithlord48) [inline doc/mainpage.md:1]: This main page should be similar to our README.md ?

I tend to think of the `readme` as the intro page to our documents at least.

File: doc/server-configs.md
Comment (nbolton) [reply doc/server-configs.md:1]: Which link?

File: doc/CMakeLists.txt (hunk 3 above)
Comment (nbolton) [reply doc/CMakeLists.txt:24]: > If you like, but I didn't touch this... I think this was from when we had `cmake-format`?

File: doc/CMakeUserPresets.json (hunk 4 above)
Comment (nbolton) [reply doc/CMakeUserPresets.json:21]: Shame JSON doesn't support comments :|

File: doc/config.toml.example (hunk 5 above)
Comment (nbolton) [reply doc/config.toml.example:1]: Ah. So, rename `doc/server-configs.md` back to `doc/configuration.md` and have it for all types of config?

File: doc/mainpage.md (hunk 6 above)
Comment (nbolton) [reply doc/mainpage.md:1]: IDK... you do your thing!

File: .gitignore [hunk 7]
```diff
@@ -26,12 +26,14 @@ deskflow-config.toml
...
 
-#CMake ignores
-CMakeLists.txt.user
-CMakeCache.txt
-CMakeUserPresets.json
-CMakeFiles/*
+# CMake files
+/CMakeLists.txt.user
+/CMakeCache.txt
+/CMakeUserPresets.json
```
Comment (sithlord48) [inline .gitignore:35]: why the / , i have never used a preceding / in other repos before.

**Comment by nbolton** [comment]
Added task:
> Add protocol compatibility for Barrier/IL to readme

File: doc/config.toml.example (hunk 5 above)
Comment (sithlord48) [reply doc/config.toml.example:1]: i would rather have it all in one spot  its all under the configuration right . figure its easier to find if its all in one document

File: .gitignore (hunk 7 above)
Comment (nbolton) [reply .gitignore:35]: It matches only files in the root. I set it like that so we can add `doc/CMakeUserPresets.json` and the rest was for consistency and clarity, but if you're against the `/` we can remove it and either force add `doc/CMakeUserPresets.json` or call that file `doc/CMakeUserPresets.json.example`.

**Comment by nbolton** [comment]
@sithlord48 You mentioned yesterday you're thinking about dropping this PR. How about dropping some of the stuff such as the hard-coded build instructions? I think there's still a lot of good stuff in here. 

I also think it's worth keeping files like `build.md` and `contribute.md` but simply as links to the relevant wiki pages, since these are common files that people look for in the source tree.

**Comment by sithlord48** [comment]
> @sithlord48 You mentioned yesterday you're thinking about dropping this PR. How about dropping some of the stuff such as the hard-coded build instructions? I think there's still a lot of good stuff in here, such as the config examples. Those could go in the wiki though, I suppose!
> 
> To confirm: you did say we should only have the build guide on the wiki and not in source, right?
> 
> Also, I think it's worth keeping files like `build.md` and `contribute.md` but simply as links to the relevant wiki pages, since these are common files that people look for in the source tree.

I don't want to drop this pr we should ship some documentation 
HOWEVER, 
  I think that the documentation we have started is not the documentation we want to ship with our deskflow installation.
We need to move the development parts of this in to relevant  Wiki pages and  work more to make these documents more user facing

I will work to this modified to reflect this change

File: README.md [hunk 8]
```diff
@@ -67,82 +67,6 @@ Here are a few ways to join in with the project and get involved:
 > Check the [Build Guide](https://github.com/deskflow/deskflow/wiki/Build-Guide)
```
Comment (nbolton) [inline README.md:67]: Let's remove the `[!TIP]` and quote (`>`), then to the new page:
```
For instructions on building Deskflow, use the wiki page: [Building](https://github.com/deskflow/deskflow/wiki/Building)
```

**Review by nbolton — APPROVED** [review]

**Pull request merged**